iOS/Flows/Home/_New/new_ConversationViewController.swift

```diff
@@ -101,35 +89,88 @@ class new_ConversationViewController: FullScreenViewController, CollectionViewIn
     }
 }
 
-extension new_ConversationViewController: UICollectionViewDelegate {
-
+extension new_ConversationViewController: UICollectionViewDelegate, UICollectionViewDelegateFlowLayout {
 
     func collectionView(_ collectionView: UICollectionView,
                         willDisplay cell: UICollectionViewCell,
                         forItemAt indexPath: IndexPath) {
 
-        // If all the messages are loaded, no need to fetch more.
+        // If all the messages are loaded, there's no need to fetch more.
         guard !self.conversationController.hasLoadedAllPreviousMessages else { return }
 
         // Start fetching new messages once the user is nearing the end of the list.
-        guard indexPath.row < 10 else { return }
+        guard indexPath.row < 2 else { return }
 
         Task {
+            guard !isLoadingMessages else { return }
+
+            self.isLoadingMessages = true
             do {
                 let oldestMessageID = self.conversationController.messages.first?.id
                 try await self.conversationController.loadPreviousMessages(before: oldestMessageID)
             } catch {
                 logDebug(error)
             }
+            self.isLoadingMessages = false
         }
     }
-}
 
-extension new_ConversationViewController {
+    func collectionView(_ collectionView: UICollectionView,
```

Comment:
Seems like all of the layout logic is hanging out here in the delegate functions which seems messy. I want to move this elsewhere soon.